cupy/linalg/solve.py

```diff
@@ -294,6 +297,80 @@ def inv(a):
     return b
 
 
+def _batched_inv(a):
+
+    assert(a.ndim >= 3)
+    a_shape = a.shape
+    a = a.copy().reshape(-1, a_shape[-2], a_shape[-1])
```

Comment:
Could you support shapes `(..., 0, 0)`?  Early-return looks good.

Reply:
I see. So you would like to avoid divide by zero, right?

    a = a.copy().reshape(-1, a_shape[-2], a_shape[-1])
    ...
    ZeroDivisionError: integer division or modulo by zero

Reply:
Is it OK to return `a`, if a shape of `a` is something like `(..., 0, 0)`?

Reply:
NumPy returns a new array.  How about `cupy.empty_like(a)` (after `a.dtype` is checked)?

Reply:
OK, will fix in that way.